packages/react/emotion-utils/src/types.ts

```diff
@@ -10,6 +10,9 @@ export type AxisDirection = 'vertical' | 'horizontal';
 export type InferenceHTMLElement<K extends keyof JSX.IntrinsicElements> = NonNullable<
   Extract<JSX.IntrinsicElements[K]['ref'], React.RefObject<any>>['current']
 >;
-export interface AsProps<T extends keyof JSX.IntrinsicElements> extends AllHTMLAttributes<T> {
+
+export type StringElementType = ElementType & string;
```

Comment:
StringElementType can restrict to ElementType can't accept ComponentType, only accept JSX.InstrincElement

```ts
type StringElementType = ElementType & string
```

```ts
type ElementType<P = any> =
  {
    [K in keyof JSX.IntrinsicElements]: P extends JSX.IntrinsicElements[K] ? K : never
  }[keyof JSX.IntrinsicElements] |
  ComponentType<P>;
```

Reply:
I thought StringElementType can be awkward name. so If you(reviewers) think so too, could you guide it to rename?